presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -381,7 +419,96 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
     {
         IcebergTableHandle handle = (IcebergTableHandle) tableHandle;
         org.apache.iceberg.Table icebergTable = getHiveIcebergTable(metastore, hdfsEnvironment, session, handle.getSchemaTableName());
-        return TableStatisticsMaker.getTableStatistics(typeManager, constraint, handle, icebergTable);
+        TableStatistics icebergStats = TableStatisticsMaker.getTableStatistics(typeManager, constraint, handle, icebergTable);
```

Comment:
> Looking through the implementation of getTableStatistics, I see that the constraint is used to limit the stats computed to only those partitions which the constraint applies to ? This is the similar to the behavior that we have [in the hive metastore](https://github.com/prestodb/presto/blob/803a534e71fb4a616f8a1b287399dd7026b8bfe3/presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java#L846-L849) implementation.

It was my intention to model this implementation like the Hive one. However, the difference between this and Hive is that we're not able to actually list out all of the partition values for the table AFAIK. Iceberg hides that behind its API. Anyways, the `filterStatsCalculatorService` doesn't necessarily filter out stats based on partitions. It filters based on the entire predicate (whether partitioned by the columns in the predicate or not)

> If so, then on line 429, we shouldn't use metastore.getTableStatistics to get the full table's stats. Instead lets use metatstore.getPartitionStatistics ?

If you look in [HiveTableOperations.java](https://github.com/prestodb/presto/blob/803a534e71fb4a616f8a1b287399dd7026b8bfe3/presto-iceberg/src/main/java/com/facebook/presto/iceberg/HiveTableOperations.java#L254-L263) you'll see that even if we create a table in presto with partitioning, we don't actually tell the Hive metastore about the partitioning since it's taken care of by Iceberg. 

In tables using the regular Hive connector that are partitioned, `get/SetPartitionStatistics` is used only when the table is partitioned inside of the [`MetastoreHiveStatisticsProvider` implementation](https://github.com/prestodb/presto/blob/803a534e71fb4a616f8a1b287399dd7026b8bfe3/presto-hive/src/main/java/com/facebook/presto/hive/statistics/MetastoreHiveStatisticsProvider.java#L121-L128). Otherwise they use `get/setTableStatistics`.

So, since the current Iceberg implementation doesn't tell Hive about the partitioning and we don't have access to partition values, I think it is safe to use the `get/setTableStatistics` methods here rather than the partition ones.

Reply:
Thanks for the explanation - I understand the split setup between HMS/Iceberg a bit better now. Given that HMS 'sees' the table as un-partitioned, while Iceberg is partitition aware and provides partition level stats (FWIU) -  I think we should be conservative (at least to start off) and only merge stats if the table is unpartitioned.

Reply:
After some more tests I understood the issue -- I've implemented this to only work for unpartitioned tables for now and added some more tests